virttest/qemu_storage.py

```diff
@@ -158,6 +158,11 @@ def create(self, params, ignore_errors=False):
                 logging.warning("We'll try to proceed by creating the dir. "
                                 "Other errors may ensue")
                 os.makedirs(image_dirname)
+            elif os.path.exists(self.image_filename):
+                # Remove previous image to avoid qemu-img attempting to get
+                # info as VM might still be running and using it which
+                # causes "write" lock issue.
+                os.unlink(self.image_filename)
```

Comment:
> either we fail to kill it or keep_between_tests is enabled

This is not a good solution to solve the problems you mentioned.  It could cause other problems  as stated by @luckyh .
To solve it:
1.  if it was failed to get killed, it is better to find out  why it was failed to get killed.
2. if` keep_between_tests` is enabled, it would be conflicts in the test case. Better fix it there, ether disable ` keep_between_tests`(shut the vm down), or  remove the image file in the test case script if it has to be using the same file name.